src/Lib/Api/fetchJoke.ts

```diff
@@ -2,14 +2,14 @@ import jokeType from '../../Types/jokeType';
 import baseUrl from './BaseUrl';
 import fetchData from './fetchData';
 
-const fetchJoke = async (name: string, category: string) => {
+const fetchJoke = async (name: string, category: string[]) => {
   let url = `${baseUrl}/random`;
-  if (name !== '' && category !== '') {
-    url += `?firstName=${name}&lastName=&limitTo=${category}`;
+  if (name !== '' && category.length !== 0) {
```

Comment:
empty string is always false, so just 'name' is enough, empty array is true, but length of empty array is 0 which means false, you can simplify this:
`if(name && !category.length)`

if your array could be undefined so the length is not available, you can make it as `array?.length`